examples/step-69/step-69.cc

```diff
@@ -0,0 +1,2926 @@
+  // use to populate parameter values: we first initialize the
+  // ParameterAcceptor class by calling its constructor with a string
+  // <code>subsection</code> denoting the correct subsection in the
+  // parameter file. Then, in the constructor body every parameter value is
+  // initialized to a sensible default value and registered with the
+  // ParameterAcceptor class with a call to
+  // ParameterAcceptor::add_parameter.
+
+  template <int dim>
+  Discretization<dim>::Discretization(const MPI_Comm &   mpi_communicator,
+                                      TimerOutput &      computing_timer,
+                                      const std::string &subsection)
+    : ParameterAcceptor(subsection)
+    , mpi_communicator(mpi_communicator)
+    , triangulation(mpi_communicator)
+    , mapping(1)
+    , finite_element(1)
+    , quadrature(3)
+    , face_quadrature(3)
+    , computing_timer(computing_timer)
+  {
+    length = 4.;
+    add_parameter("immersed disc - length",
+                  length,
+                  "Immersed disc: length of computational domain");
+
+    height = 2.;
+    add_parameter("immersed disc - height",
+                  height,
+                  "Immersed disc: height of computational domain");
+
+    disc_position = 0.6;
+    add_parameter("immersed disc - object position",
+                  disc_position,
+                  "Immersed disc: x position of immersed disc center point");
+
+    disc_diameter = 0.5;
+    add_parameter("immersed disc - object diameter",
+                  disc_diameter,
+                  "Immersed disc: diameter of immersed disc");
+
+    refinement = 5;
+    add_parameter("initial refinement",
+                  refinement,
+                  "Initial refinement of the geometry");
+  }
+
+  // Note that in the previous constructor we only passed the MPI
+  // communicator to the <code>triangulation</code> but we still have not
+  // initialized the underlying geometry/mesh. As mentioned earlier, we
+  // have to postpone this task to the <code>setup()</code> function that
+  // gets called after the ParameterAcceptor::initialize() function has
+  // populated all parameter variables with the final values read from the
+  // parameter file.
+  //
+  // The <code>setup()</code> function is the last class member that has to
+  // be implemented. It creates the actual triangulation that is a
+  // benchmark configuration consisting of a channel with a disc obstacle, see
+  // @cite GuermondEtAl2018. We construct the geometry by modifying the
+  // mesh generated by GridGenerator::hyper_cube_with_cylindrical_hole().
+  // We refer to Step-49, Step-53, and Step-54 for an overview how to
+  // create advanced meshes.
+
+  template <int dim>
+  void Discretization<dim>::setup()
+  {
+    TimerOutput::Scope t(computing_timer, "discretization - setup");
+
+    triangulation.clear();
+
+    // We first create 4 temporary (non distributed) coarse triangulations
```

Comment:
Move this comment to the one above the function. When you read through the html of tutorial programs, it turns out that things get difficult to read if the code is broken up into too many smaller pieces. It's often better if one can see the bigger picture, and I think here it would be useful to show everything from the beginning of the function down to the call to `merge_triangulations()` in the same block.